app/services/db/monthly_sp_auth_count/unique_monthly_auth_counts_by_partner.rb

```diff
@@ -0,0 +1,144 @@
+
+          year_months.each do |year_month|
+            year_month_users = year_month_to_users[year_month]
+
+            auth_count = year_month_users.count
+            unique_users = year_month_users.uniq.to_set
+
+            new_unique_users = unique_users - prev_seen_users
+            new_unique_users_year1 = new_unique_users.select { |user_id, profile_age| profile_age == 0 }
+            new_unique_users_year2 = new_unique_users.select { |user_id, profile_age| profile_age == 1 }
+            new_unique_users_year3 = new_unique_users.select { |user_id, profile_age| profile_age == 2 }
+            new_unique_users_year4 = new_unique_users.select { |user_id, profile_age| profile_age == 3 }
+            new_unique_users_year5 = new_unique_users.select { |user_id, profile_age| profile_age == 4 }
+            new_unique_users_year_greater_than_5 = new_unique_users.select { |user_id, profile_age| profile_age > 4 }
+            new_unique_users_unknown = new_unique_users.select { |user_id, profile_age| profile_age < 0 }
+            prev_seen_users |= unique_users
+
+            rows << {
+              key: key,
+              ial: ial,
+              year_month: year_month,
+              iaa_start_date: date_range.begin.to_s,
+              iaa_end_date: date_range.end.to_s,
+              unique_users: unique_users.count,
+              new_unique_users: new_unique_users.count,
+              partner_ial2_new_unique_users_year1: new_unique_users_year1.count,
+              partner_ial2_new_unique_users_year2: new_unique_users_year2.count,
+              partner_ial2_new_unique_users_year3: new_unique_users_year3.count,
+              partner_ial2_new_unique_users_year4: new_unique_users_year4.count,
+              partner_ial2_new_unique_users_year5: new_unique_users_year5.count,
+              partner_ial2_new_unique_users_year_greater_than_5: new_unique_users_year_greater_than_5.count,
+              partner_ial2_new_unique_users_unknown: new_unique_users_unknown.count,
+            }
+          end
+        end
+
+        rows
+      end
+
+      # @param [Array<String>] issuers all the issuers for this iaa
+      # @param [Array<Range<Date>>] months ranges of dates by month that are included in this iaa,
+      #  the first and last may be partial months
+      # @return [Array<String>]
+      def build_queries(issuers:, months:)
+        months.map do |month_range|
+          params = {
+            range_start: month_range.begin,
+            range_end: month_range.end,
+            year_month: month_range.begin.strftime('%Y%m'),
+            issuers: issuers,
+          }.transform_values { |value| quote(value) }
+
+
+
+          format(<<~SQL, params)
+            SELECT
+              sp_return_logs.user_id
+            , %{year_month} AS year_month
+            , sp_return_logs.ial
+            , MIN(sp_return_logs.returned_at) AS min_returned_at
+            , MAX(sp_return_logs.profile_verified_at) AS max_profile_verified_at
```

Comment:
hmmm looking at this... I'm wondering if maybe we want to find a way to do the age calculation per-row of `sp_return_logs` (ex look at the age of the profile that was used at the time it was used)


Like maybe we push this logic into a subquery and then do a `MIN` on that? 

```
SELECT
  subq.user_id
, subq.ial
, MIN(profile_years_old)
FROM (
  SELECT
    DATE_PART('year', AGE(sp_return_logs.returned_at, sp_return_logs.profile_verified_at) as profile_years_old
  FROM
    sp_return_logs
) subq
```